packages/api-derive/src/contracts/fees.ts

```diff
@@ -6,43 +6,114 @@ import { ApiInterface$Rx } from '@polkadot/api/types';
+ * all relevant contract fees as declared in the substrate chain spec.
+ * @example
+ * <BR>
+ *
+ * ```javascript
+ * api.derive.contracts.fees(([creationFee, transferFee]) => {
+ *   // console.log(`The fee for creating a new contract on this chain is ${creationFee} units. The fee required to call this contract is ${transferFee} units.`);
+ * });
+ * ```
  */
 export function fees (api: ApiInterface$Rx) {
   return (): Observable<DerivedContractFees> => {
     const queryBase = api.query.contracts || api.query.contract;
 
-    // FIXME A numnber of these are being moved https://github.com/paritytech/substrate/pull/2883
-    return (api.queryMulti([
-      queryBase.callBaseFee,
-      queryBase.contractFee,
-      queryBase.createBaseFee,
-      queryBase.creationFee,
-      queryBase.rentByteFee,
-      queryBase.rentDepositOffset,
-      queryBase.transactionBaseFee,
-      queryBase.transactionByteFee,
-      queryBase.transferFee,
-      queryBase.tombstoneDeposit
-    ]) as any as Observable<Array<BN>>).pipe(
-      map(([callBaseFee, contractFee, createBaseFee, creationFee, rentByteFee, rentDepositOffset, transactionBaseFee, transactionByteFee, transferFee, tombstoneDeposit]) => ({
-        callBaseFee,
-        contractFee,
-        createBaseFee,
-        creationFee,
-        rentByteFee,
-        rentDepositOffset,
-        transactionBaseFee,
-        transactionByteFee,
-        transferFee,
-        tombstoneDeposit
-      } as DerivedContractFees)),
+    if (api.query.contract && !api.query.contract.rentByteFee) {
+      // Only for 1.0 support. rentByteFee, rentDepositOffset, tombstoneDeposit are not available in substrate 1.0.
+      // @TODO remove this once 1.0 support is dropped
+      const ZERO = new BN(0);
+      return (combineLatest([
+        of([ZERO, ZERO, ZERO]),
+        api.queryMulti([
+          queryBase.callBaseFee,
+          queryBase.contractFee,
+          queryBase.createBaseFee,
+          queryBase.creationFee,
+          queryBase.transactionBaseFee,
+          queryBase.transactionByteFee,
+          queryBase.transferFee
+        ])
+      ]) as any as Observable<[[BN, BN, BN], [BN, BN, BN, BN, BN, BN, BN]]>).pipe(
+        map(([[
+            rentByteFee,
+            rentDepositOffset,
+            tombstoneDeposit
+          ],[
+            callBaseFee,
+            contractFee,
+            createBaseFee,
+            creationFee,
+            transactionBaseFee,
+            transactionByteFee,
+            transferFee
+          ]]) => ({
+            callBaseFee,
+            contractFee,
+            createBaseFee,
+            creationFee,
+            rentByteFee,
+            rentDepositOffset,
+            tombstoneDeposit,
+            transactionBaseFee,
+            transactionByteFee,
+            transferFee
+          } as DerivedContractFees)),
+        drr()
+      );
+    }
+
+    return (api.consts.contracts
```

Comment:
Same as the readability ni comment above, i.e.

```js
return (
  api.consts.contracts
    ? of([..])
    : api.queryMulti([...])
)
```

Reply:
changed